Test Variations of the Eclipse Icon

Michael5601 opened the pull request:
This draft is created in the context of this [issue](https://github.com/eclipse-platform/eclipse.platform.images/issues/153#issue-3178350003) and allows analyzing different variations of the Eclipse Icon. 

How to test:
I added three different variations as SVG of the eclipse icon  in different sizes to the bundle `org,eclipse.sdk`:

- `eclipse16.svg`: Normal eclipse icon as SVG. After rendering + antialiasing the white lines in the middle of the eclipse icon look like one white block.
- `eclipse16_1.0_space.svg`: Variation of the eclipse icon where the white space and the blue lines in the middle of the icon each are 1.0px high.
- `eclipse16_1.25_space.svg`: Same as above but lines and white space are 1.25px high.

To change the currently used icons the `plugin.xml` of the bundle `org.eclipse.sdk` needs to be adjusted. The default of this draft is the 1.0px variation. The icons can be found in the top left corner of the window or in the about dialog found in the toolbar entry `Help` -> `About Eclipse IDE`. 

Screenshots can be found in the above cited issue.

BeckerWdf commented:
> * `eclipse16_1.0_space.svg`: Same as above but lines and white space are 1.25px high.

I think you mean `eclipse16_1.25_space.svg`

BeckerWdf commented:
> The icons can be found in the top left corner of the window

But only on windows. macOS windows don't have a program icons.

Michael5601 commented:
> > * `eclipse16_1.0_space.svg`: Same as above but lines and white space are 1.25px high.
> 
> I think you mean `eclipse16_1.25_space.svg`

yes I was a copy error, thanks.

Michael5601 commented:
> > The icons can be found in the top left corner of the window
> 
> But only on windows. macOS windows don't have a program icons.

ah this was new to me. Good to know.

HeikoKlare reviewed (commented):
I have started testing the proposed icons at 100% zoom. Unfortunately, both versions don't look good to me. The 1.0px space version looks blurry, whereas in the 1.25px space version the spaces look to large to me (compared to the original icon).

The following screenshots show the existing PNG icon to the left and the SVG icon to the right.

1.0px_space:
![image](https://github.com/user-attachments/assets/d18e61a3-9cc1-4902-9783-fdcf0d71180e)

1.25px_space:
![image](https://github.com/user-attachments/assets/1ef3436c-5aa4-4326-8422-0218e37965cb)

When taking a look at the SVG (of the 1.0px_space version), I see that the white lines are placed in the middle of a pixel, leading to blurry rasterization due to anti aliasing. When I move the lines to fit into pixel it looks better to me:
![image](https://github.com/user-attachments/assets/f5852cae-2de7-4775-9894-59a34ab9bcb5)

Michael5601 commented:
Thanks for testing @HeikoKlare. As I stated before I also saw the problems with 1.0px and 1.25px and proposed to additionally try 1.15px but your solution to align the 1.0px lines pixel perfect looks even better. 

With this information I propose to add the Eclipse Icon as SVG to the Plugins with the following specification:
White space and blue lines each 1.0px in height but alligned pixel perfect as can be seen at the bottom of your comment.

BeckerWdf commented:
> When I move the lines to fit into pixel it looks better to me

Yes. It's important not align horizontal and vertical lines at exact pixel borders.

BeckerWdf commented:
> With this information I propose to add the Eclipse Icon as SVG to the Plugins with the following specification:
> White space and blue lines each 1.0px in height but alligned pixel perfect as can be seen at the bottom of your comment.

Yes that sounds good.

Michael5601 commented:
Thank your for the input @HeikoKlare and @BeckerWdf.
I changed the icons according to the result of this discussion and added them to this [PR](https://github.com/eclipse-platform/eclipse.platform/pull/1998). Thus I will close this draft.
